test/test_jit.py

```diff
@@ -1410,30 +1406,50 @@ def forward(self, x):
                 rq = torch.quantize_per_tensor(r, 0.2, 1, torch.quint8)
                 return rq
 
+        class QConv(torch.nn.Module):
+            def __init__(self):
+                super(QConv, self).__init__()
+                self.weight = torch.nn.Parameter(torch.rand((2, 3, 3, 3), dtype=torch.float))
+                self.bias = torch.nn.Parameter(torch.rand((2,), dtype=torch.float))
 
-        m = torch.jit.script(M())
-        data = torch.randn((5, 5), dtype=torch.float)
-        ref_res = m._c._get_method('forward')(data)
-        torch._C._jit_pass_fold_prepack(m._c, torch.jit.script(PackedParams())._c)
-        res = m._c._get_method('forward')(data)
-        # check attribute and graph
-        self.assertTrue(m._c._has_module('_packed_linear_weight_bias'))
-        # check values
-        original_w = m._c._get_parameter('weight')
-        ref_w = torch.quantize_per_tensor(original_w, 0.2, 1, torch.qint8).dequantize()
-        ref_b = m._c._get_parameter('bias')
-        w, b = m._c._get_module('_packed_linear_weight_bias')._get_method('_weight_bias')()
-        self.assertEqual(ref_w, w.dequantize())
-        self.assertEqual(ref_b, b)
-        self.assertEqual(ref_res, res)
-
-        # test serialization
-        buffer = io.BytesIO()
-        torch.jit.save(m, buffer)
-        buffer.seek(0)
-        loaded_mod = torch.jit.load(buffer)
-        loaded_res = loaded_mod._c._get_method('forward')(data)
-        self.assertEqual(ref_res, loaded_res)
+            def forward(self, x):
+                xq = torch.quantize_per_tensor(x, 0.2, 2, torch.quint8)
+                wq = torch.quantize_per_tensor(self.weight, 0.2, 2, torch.qint8)
+                stride, padding, dilation, groups = [1, 1], [0, 0], [1, 1], 1
+                packed = torch.ops.quantized.conv_prepack(wq, self.bias, stride, padding, dilation, groups)
+                w_unpacked, b_unpacked = torch.ops.quantized.conv_unpack(packed)
+                r = torch.nn.functional.conv2d(xq.dequantize(), w_unpacked.dequantize(), b_unpacked, stride, padding, dilation, groups)
+                rq = torch.quantize_per_tensor(r, 0.2, 2, torch.quint8)
+                return rq
+
+        for name, M, data in [('linear', QLinear, torch.randn((5, 5), dtype=torch.float)),
+                              ('conv', QConv, torch.randn((1, 3, 24, 24), dtype=torch.float))]:
+            m = torch.jit.script(M())
+            ref_res = get_forward(m._c)(data)
+            fold_prepack(m)
```

Comment:
yeah, you mean split the check in 1441 into a different file? but the behavior we want is these weights are packed properly in this context. I think there is still value to put this here

Reply:
I meant testing the functionality of these classes directly without using our pass (e.g. using the class directly in Python it to pack, then unpack and verifying that the values are the same, maybe also packing and making sure the packed values are accepted by quantized kernels).

Reply:
I think this is more about checking the result of fold_prepack rather than test the functionality of the PackedParams modules. basically we want to make sure we insert the module and called `set_weight_bias` on correct values.